### DISCUSSION TITLE
triggers : kubernetes-workload

### DISCUSSION THREAD

**Discussion by jitender-bst** [kedacore/keda#2919] [Q&A / Need Help]
I was exploring Keda and tried to get started with the kubernetes-workload trigger , firstly  I installed keda in the cluster via helm and then I created 2 deployment one  with name testing and another with name testing1 . As my per desire if the testing deployment no. of pods got increased the pods from the testing1 deployment should also scale .

**for the scaled objects I used the below yaml :**

> apiVersion: keda.sh/v1alpha1
> kind: ScaledObject
> metadata:
>   name: workload-scaledobject
> spec:
>   scaleTargetRef:
>     name: **testing1**
>   triggers:
>   - type: kubernetes-workload
>     metadata:
>       podSelector: '**app=testing**'
>       value: '1'

what when i scaled the pods from testing deployment the no. of pods for testing1 remains same .

Output of **kubectl get so** : 
<img width="1723" alt="image" src="https://user-images.githubusercontent.com/97177383/163993848-be0b364a-7025-4f15-a4de-53d82f6b5b46.png">

output of **kubectl describe so workload-scaledobject** : 
<img width="1721" alt="image" src="https://user-images.githubusercontent.com/97177383/163994046-d2db28e8-8b31-4cf2-a753-a97d1fd26456.png">

<img width="1606" alt="image" src="https://user-images.githubusercontent.com/97177383/163994130-55039392-ef5d-4848-a0a8-d376610eefa3.png">

**

**I can see error in the HPA though :**
** 
<img width="1726" alt="image" src="https://user-images.githubusercontent.com/97177383/163994347-306e0303-9dd5-43b3-8148-f47f314a87ad.png">

can anybody help me out from this please.

**Comment by tomkerkhove** [comment]
Are you able to use other scalers? It might be that the HPA is not able to fetch the metrics but think it would be easier if we could see the KEDA logs.

**Reply by jitender-bst** [reply]
Hi ,

have checked the keda logs via the command **kubectl logs keda-operator-778cf49bcf-28jp5  -c keda-operator** : 

but I cannot see any kind of error in it .

<img width="1723" alt="image" src="https://user-images.githubusercontent.com/97177383/164038665-1b7a57fb-2ab6-4315-9666-c776ad00713f.png">

**Reply by zroubalik** [reply]
I think the problem is in KEDA Metrics Server. Do you see any errors there? Also, what platfrom are you deploying to?

**Reply by jitender-bst** [reply]
Can see a error related to metric server in HPA and it says : 
<img width="1726" alt="image" src="https://user-images.githubusercontent.com/97177383/164153320-a4bfe2a5-5009-43cd-bca4-94633e84f5a0.png">

I am using GKE for this POC.

I would also like to tell one strange behaviour that is when I put wrong value for the podselector , it scales down my scaleTargerref deployment to zero and when I correct the value for podselector it scales up to 1 pod.

**Reply by tomkerkhove** [reply]
A-ha, GKE.

Then you might want to check our troubleshooting guide - https://keda.sh/docs/latest/troubleshooting/
